packages-web/maps/src/components/LeafletMap.tsx

```diff
@@ -0,0 +1,186 @@
+import { createElement, ReactElement, useEffect, useRef, useState } from "react";
+import {
+    FeatureGroup,
+    LatLngLiteral,
+    Map,
+    MapOptions,
+    Marker as LeafletMarker,
+    icon,
+    tileLayer,
+    TileLayerOptions,
+    TileLayer
+} from "leaflet";
+import { MapProviderEnum, Marker, SharedProps } from "../../typings";
+import { Alert } from "@widgets-resources/piw-utils";
+import classNames from "classnames";
+import { customUrls, getDimensions, mapAttr } from "../utils";
+
+export interface LeafletProps extends SharedProps {
+    mapProvider: MapProviderEnum;
+    attributionControl: boolean;
+}
+
+export const LeafletMap = (props: LeafletProps): ReactElement => {
+    const map = useRef<Map>();
+    const leafletRef = useRef<HTMLDivElement>(null);
+    const defaultCenterLocation: LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
+    const markerGroup = useRef<FeatureGroup>(new FeatureGroup());
+
+    const [validationMessage, setValidationMessage] = useState(props.validationMessage);
+
+    useEffect(() => {
+        createMap();
+
+        return () => {
+            if (map.current) {
+                map.current.off();
+                map.current.remove();
+            }
+        };
+    }, []);
+
+    useEffect(() => {
+        addMarkers();
+    }, [props.locations, props.currentLocation]);
+
+    const createMap = (): void => {
+        const mapOptions: MapOptions = {
+            scrollWheelZoom: props.optionScroll,
+            zoomControl: props.optionZoomControl,
+            attributionControl: props.attributionControl,
+            zoom: props.zoomLevel,
+            minZoom: 1,
+            maxZoom: 18, // Max zoom for HERE, OSM & MapBox
+            dragging: props.optionDrag,
+            center: defaultCenterLocation,
+            closePopupOnClick: false
+        };
+        if (leafletRef.current && !map.current) {
+            map.current = new Map(leafletRef.current, mapOptions)
+                .addLayer(baseMapLayer())
+                .addLayer(markerGroup.current);
+        }
+        addMarkers();
+    };
+
+    const addMarkers = (): void => {
+        markerGroup.current.clearLayers();
+        if (props.locations && props.locations.length) {
+            props.locations.forEach(location => {
+                markerGroup.current.addLayer(addMarker(location));
+            });
+        }
+        addCurrentLocation();
+        updateCamera();
+    };
+
+    const addMarker = (marker: Marker): LeafletMarker => {
+        const mapMarker = new LeafletMarker([Number(marker.latitude), Number(marker.longitude)], {
+            title: marker.title,
+            alt: marker.title
+        });
+        if (marker.title) {
+            mapMarker.setTooltipContent(marker.title);
+            if (marker.onClick) {
+                const infoContent = document.createElement("span");
+                infoContent.innerHTML = marker.title || "";
+                if (marker.onClick) {
```

Comment:
I dont think we need to check onclick here again since its checked in line 84

Reply:
Done